sdk/resourcemanager/Azure.ResourceManager/docs/MigrationGuide.md

```diff
@@ -0,0 +1,287 @@
+# Migrating from old to preview management SDK.
+
+There are several differences between the old sdk and this preview. Here's an example of how to create a Virtual Machine with both SDKs:
+
+## Create a Virtual Machine example
+
+### Import the namespaces
+#### Old (Microsoft.Azure.Management._)
+```C#
+using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using Microsoft.Azure.Management.ResourceManager;
+using Microsoft.Rest;
+using System;
+using System.Threading.Tasks;
+```
+#### New (Azure.Azure.ResourceManager._ Preview)
+```C#
+using Azure.Identity;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.Compute;
+using Azure.ResourceManager.Network;
+using System;
+using System.Threading.Tasks;
+```
+
+### Setting up the clients
+#### Old
+```C#
+ServiceClientCredentials credentials = getMyCredentials();
+var computeClient = new ComputeManagementClient(credentials);
+var networkClient = new NetworkManagementClient(credentials);
+var managedServiceIdentityClient = new ManagedServiceIdentityClient(credentials);
+```
+#### New
+```C#
+var armClient = new ArmClient(new DefaultAzureCredential());
+```
+As you can see, authentication is now handled by Azure.Identity, and now just a single client is needed, from which you can get the `DefaultSubscription` and start managing your resources. 
+
+### Create a Resource Group
+#### Old
+```C#
+ServiceClientCredentials credentials = getMyCredentials(); 
+var resourcesClient = new ResourceManagementClient(credentials);
+
+string rgName = "QuickStartRG";
+string location = "WestUS2";
+resourcesClient.ResourceGroups.CreateOrUpdate(
+    rgName,
+    new ResourceGroup
+    {
+        Location = location,
+        Tags = new Dictionary<string, string>() { { rgName, DateTime.UtcNow.ToString("u") } }
+    });
+```
+#### New
+```C#
+var armClient = new ArmClient(new DefaultAzureCredential());
+Subscription subscription = armClient.DefaultSubscription;
+ResourceGroupContainer rgContainer = subscription.GetResourceGroups();
+
+Location location = Location.WestUS2;
+string rgName = "QuickStartRG";
+ResourceGroup resourceGroup = await rgContainer.Construct(location).CreateOrUpdateAsync(rgName)
```

Comment:
Construct is removed for preview call Create directly and pass in a new ResourceGroupData(...)